aten/src/ATen/functorch/LegacyBatchingRegistrations.cpp

```diff
@@ -704,6 +705,19 @@ Tensor new_empty_strided_batching_rule(
   return physical_view.getPhysicalToLogicalMap().apply(result);
 }
 
+Tensor nested_tensor_size_batching_rule(const Tensor& self) {
+  auto* maybe_batched_impl = maybeGetBatchedImpl(self);
+  if (!maybe_batched_impl) {
+    TORCH_INTERNAL_ASSERT(false,
+        "Tried to run batching rule for _nested_tensor_size() on a non-batched tensor")
+  }
+  auto* nt_impl = at::native::get_nested_tensor_impl(maybe_batched_impl->value());
+  if (!nt_impl) {
+    TORCH_INTERNAL_ASSERT(false,
+        "Tried to run batching rule for _nested_tensor_size() on a non-nested tensor")
+  }
+  return nt_impl->get_nested_sizes();
+}
```

Comment:
> Are we limiting the scope of this to vmap over NestedTensor where only the nested tensor's first dimension is batched?

Yes, I've been operating off of this assumption, since NT has historically been defined as a batch of possibly ragged things, and dim=0 is required to be the batch dim. I also think this simplifies vmap support a decent amount from what I can tell.

> If so, I would expect ._nested_sizes() to raise an error, because from the POV of the original code, we're not actually dealing with any nested tensors.

Interesting, I think this may or may not be the case. You could imagine a fallback that runs on NT slices of NTs (more similar to what the current fallback does) rather than on the fully unbound components.

That said, I added this `_nested_tensor_size()` support mostly for debugging, and it is a private API. I really don't expect people to validly be calling this from within their code. So I can drop this before merging, no problem there.